tasks/scrapeData/dedupData.js

```diff
@@ -0,0 +1,113 @@
+import * as geography from '../../lib/geography.js';
+
+const numericalValues = ['cases', 'tested', 'recovered', 'deaths', 'active'];
+
+/*
+  Returns a report if the crosscheck fails, or false if the two sets have identical data
+*/
+function crosscheck(a, b) {
+  const crosscheckReport = {};
+  let failed = false;
+  for (const prop of numericalValues) {
+    if (a[prop] !== b[prop]) {
+      crosscheckReport[prop] = [a[prop], b[prop]];
+      failed = true;
+    }
+  }
+  return failed ? crosscheckReport : false;
+}
+
+/*
+  Remove "private" object properties
+*/
+function removePrivate(data) {
+  for (const [prop, value] of Object.entries(data)) {
+    if (value === '') {
+      delete data[prop];
+    }
+    // Remove "private" fields
+    if (prop[0] === '_') {
+      delete data[prop];
+    }
+  }
+
+  return data;
+}
+
+/*
+  Check if the passed stripped location object exists in the crosscheck report
+*/
+function existsInCrosscheckReports(location, crosscheckReportsByLocation) {
+  let exists = false;
+  for (const existingLocation of crosscheckReportsByLocation) {
+    if (existingLocation.url === location.url) {
+      exists = true;
+      break;
+    }
+  }
+  return exists;
+}
+
+/*
+  Get the priority of a location
+*/
+const getPriority = location => {
+  return location.priority !== undefined ? location.priority : 0;
+};
+
+const dedupLocations = args => {
+  const { locations } = args;
+
+  const crosscheckReports = {};
+  const seenLocations = {};
+  let i = locations.length;
+  let deDuped = 0;
+  while (i-- > 0) {
+    const location = locations[i];
+    const locationName = geography.getName(location);
+    const otherLocation = seenLocations[locationName];
+
+    if (otherLocation) {
+      // Take rating into account to break ties
+      const thisPriority = getPriority(location) + location.rating / 2;
+      const otherPriority = getPriority(otherLocation) + otherLocation.rating / 2;
+
+      if (otherPriority === thisPriority) {
+        console.log('⚠️  %s: Equal priority sources choosing %s (%d) over %s (%d) arbitrarily', locationName, location.url, thisPriority, otherLocation.url, otherPriority);
+        // Kill the other location
```

Comment:
Potentially delete is a nicer word since we're talking about viruses 🤷‍♂

Reply:
Oh snap, good call. That was my wording.